sources/packages/backend/apps/api/src/services/disbursement-schedule/_tests_/unit/disbursement-schedule.getCOEApprovalPeriodStatus.spec.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  NotificationActionsService,
+  SequenceControlService,
+} from "@sims/services";
+import { DisbursementScheduleService } from "../../disbursement-schedule-service";
+import { DataSource } from "typeorm";
+import { addDays } from "@sims/utilities";
+import { COE_WINDOW } from "../../../../utilities";
+import { COEApprovalPeriodStatus } from "../../disbursement-schedule.models";
+
+describe("DisbursementScheduleService-getCOEApprovalPeriodStatus", () => {
+  let service: DisbursementScheduleService;
+
+  beforeAll(async () => {
+    const dataSource = {} as DataSource;
+    dataSource.getRepository = jest.fn();
+    const sequenceService = {} as SequenceControlService;
+    const notificationActionsService = {} as NotificationActionsService;
+    service = new DisbursementScheduleService(
+      dataSource,
+      sequenceService,
+      notificationActionsService,
+    );
+  });
+
+  it("Should throw an error when the disbursement date is undefined.", () => {
+    // Arrange
+    const disbursementDate = undefined;
+    const studyEndDate = new Date();
+
+    // Act and Assert.
+    expect(() => {
+      service.getCOEApprovalPeriodStatus(disbursementDate, studyEndDate);
+    }).toThrow(
+      "disbursementDate and studyEndDate are required for COE window verification.",
+    );
+  });
+
+  it("Should throw an error when both the disbursement date and study end date are undefined.", () => {
+    // Arrange
+    const disbursementDate = undefined;
+    const studyEndDate = undefined;
+
+    // Act and Assert.
+    expect(() => {
+      service.getCOEApprovalPeriodStatus(disbursementDate, studyEndDate);
+    }).toThrow(
+      "disbursementDate and studyEndDate are required for COE window verification.",
+    );
+  });
+
+  it(`Should return "${COEApprovalPeriodStatus.WithinApprovalPeriod}" when enrolment now is the first day of (within) the allowed approval period.`, () => {
```

Comment:
what does enrolment now mean?

Reply:
its today's enrollment, I used` enrolment now`  as it was term used in the code comment
![image](https://user-images.githubusercontent.com/77353155/222296255-d0122148-ea34-49a0-b04b-e0a7a404fd88.png)

Reply:
As per the code comment, it means the Is the enrolment now(at present at current date time) or in other words Is the enrolment at present within eligible period.

Looking at the test case comment the "enrolment now" wasn't giving same meaning IMO which made me confused.

Reply:
@andrewsignori-aot  @andrepestana-aot you guys okay with `enrolment at present`?